packages/common/src/downloads/build-existing-exports-portal-query.ts

```diff
@@ -41,12 +43,57 @@ export const PORTAL_EXPORT_TYPES = {
   },
 };
 
-interface ExistingExportsPortalQueryOptions {
+interface IExistingExportsPortalQueryOptions {
   layerId?: number | string;
   onlyTypes?: string[];
   spatialRefId?: string;
 }
 
+/**
+ * Puts a spatial reference into a serialized format that can be used
+ * for item typeKeywords.
+ *
+ * **Note**: discards "latestWkid"
+ *
+ * In the past we used `JSON.stringify`, but that causes problems because
+ * it can include commas which are interpreted by the portal [update item call](https://developers.arcgis.com/rest/users-groups-and-items/update-item.htm)
+ * as being separate typekeywords. With `JSON.stringify`, equality was also
+ * dependent on the order of the properties in the spatial reference.
+ *
+ * Check https://developers.arcgis.com/web-map-specification/objects/spatialReference/
+ * for more details on what this object looks like.
+ */
+export function serializeSpatialReference(
+  spatialReference: ISpatialReference | string
+): string {
+  if (typeof spatialReference === "object") {
+    const { wkid, wkt } = spatialReference;
+    return wkid ? wkid + "" : btoa(wkt);
+  } else {
+    return spatialReference;
+  }
+}
```

Comment:
Here's the magical serialization logic. We know that a spatialReference always contains either `wkid` (no commas) or `wkt` (possibly commas). When there are possibly commas, we base64-encode.